internal/storage/cifs/cifs.go

```diff
@@ -0,0 +1,102 @@
+package cifs
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"unicode"
+
+	"github.com/Microsoft/opengcs/internal/network"
+	"github.com/Microsoft/opengcs/internal/oc"
+	"github.com/pkg/errors"
+	"github.com/vishvananda/netns"
+	"go.opencensus.io/trace"
+	"golang.org/x/sys/unix"
+)
+
+const cifsMountOptions = "cache=loose,actimeo=10000,dir_mode=0777,file_mode=0777,sec=ntlmssp,mfsymlinks,ro,async"
+
+var (
+	osMkdirAll  = os.MkdirAll
+	osRemoveAll = os.RemoveAll
+	unixMount   = unix.Mount
+)
+
+// Mount creates a cifs mount at `target`. If `target` doesn't exist this call
+// will try and create it.
+//
+// The UVM generally doesn't have a network adapter in the default network namespace so this function
+// will lock the current goroutines thread, join a network namespace with an available adapter and then
+// perform the mount.
+func Mount(ctx context.Context, source, target, username, password string) (err error) {
+	_, span := trace.StartSpan(ctx, "cifs.Mount")
+	defer span.End()
+	defer func() { oc.SetSpanStatus(span, err) }()
+
+	span.AddAttributes(
+		trace.StringAttribute("target", target),
+		trace.StringAttribute("source", source))
+
+	if err := osMkdirAll(target, 0700); err != nil {
+		return err
+	}
+	defer func() {
+		if err != nil {
+			osRemoveAll(target)
+		}
+	}()
+
+	cmd := exec.Command(
+		"mount",
+		"-t",
+		"cifs",
+		source,
+		target,
+		"-o",
+		fmt.Sprintf("vers=3.0,username=%s,password=%s,%s", username, password, cifsMountOptions),
```

Comment:
Why can't we use the storage `Mount` function or otherwise call unix.Mount here?

Reply:
We likely can, there was some reason the teleport folks shell out to the mount binary but I need to follow up with them again for this.

Reply:
Ah I see, we issue this cmd to the network ns.

Reply:
So with the new changes you made for `DoInNetNS`, we can construct a function here that makes use of the storage `Mount` call or unix.Mount.

Reply:
That shouldn't matter for this, the whole operation either via direct syscall or invoking this should all happen in the same netns. The teleport folks don't have this restriction and they invoke the binary, I just need to follow-up again for why.

Reply:
> So with the new changes you made for `DoInNetNS`, we can construct a function here that makes use of the storage `Mount` call or unix.Mount.

Yep